src/pymodaq/utils/managers/preset_manager.py

```diff
@@ -169,8 +170,24 @@ def show_preset(self):
             # save managers parameters in a xml file
             # start = os.path.split(os.path.split(os.path.realpath(__file__))[0])[0]
             # start = os.path.join("..",'daq_scan')
-            ioxml.parameter_to_xml_file(
-                self.preset_params, os.path.join(path, self.preset_params.child('filename').value()))
+            filename_without_extension = self.preset_params.child('filename').value()
+
+            try:
+                ioxml.parameter_to_xml_file(self.preset_params,
+                                            os.path.join(path, filename_without_extension),
+                                            overwrite=False)
+            except FileExistsError as currenterror:
+                logger.warning(str(currenterror)+"File " + filename_without_extension + ".xml exists")
+                userchoice = dialogbox(title='Overwrite confirmation',
+                                       message="File exist do you want to overwrite it ?")
+                if userchoice == 1:
+                    ioxml.parameter_to_xml_file(self.preset_params,
+                                                os.path.join(path, filename_without_extension))
+                    logger.warning("File " + filename_without_extension + ".xml overwriten at user request")
```

Comment:
construct complex string using the f'' syntax:

f"File {filename_without_extension}.xml overwriten at user request"